core/src/main/java/org/apache/iceberg/jdbc/JdbcClientPool.java

```diff
@@ -43,8 +65,18 @@ public JdbcClientPool(String dbUrl, Map<String, String> props) {
   }
 
   public JdbcClientPool(int poolSize, String dbUrl, Map<String, String> props) {
-    super(poolSize, SQLNonTransientConnectionException.class, true);
+    super(poolSize, SQLTransientException.class, true);
     properties = props;
+    retryableStatusCodes = Sets.newHashSet();
+    retryableStatusCodes.addAll(COMMON_RETRYABLE_CONNECTION_SQL_STATES);
+    String configuredRetryableStatuses = props.get(JdbcUtil.RETRYABLE_STATUS_CODES);
```

Comment:
Thanks to make it configuration (as discussed in the PR): it will allow users to define database specific states.